tests/integration/test_logcollector/test_macos/test_macos_file_status_basic.py

```diff
@@ -0,0 +1,164 @@
+from wazuh_testing.tools.monitoring import FileMonitor, wait_file, make_callback
+from wazuh_testing.tools.configuration import load_wazuh_configurations
+from wazuh_testing.logcollector import prefix as logcollector_prefix
+from wazuh_testing.tools.file import read_json, truncate_file
+from wazuh_testing.tools.services import control_service
+
+# Marks
+pytestmark = [pytest.mark.darwin, pytest.mark.tier(level=0)]
+
+# Configuration
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_macos_file_status_basic.yaml')
+
+parameters = [{'ONLY_FUTURE_EVENTS': 'yes'}, {'ONLY_FUTURE_EVENTS': 'no'}]
+metadata = [{'only-future-events': 'yes'}, {'only-future-events': 'no'}]
+
+# Configuration data
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+configuration_ids = [f"{x['ONLY_FUTURE_EVENTS']}" for x in parameters]
+
+daemons_handler_configuration = {'daemons': ['wazuh-logcollector'], 'ignore_errors': False}
+
+# Max number of characters to be displayed in the log's debug message
+sample_log_length = 100
+# Time in seconds to update the file_status.json
+file_status_update_time = 4
+
+local_internal_options = { 'logcollector.debug': 2,
+                           'logcollector.vcheck_files': file_status_update_time,
+                           'logcollector.sample_log_length': sample_log_length }
+
+macos_message = {   'command': 'logger',
+                    'message': 'Logger testing message - file status'   }
+
+# Maximum waiting time in seconds to find the logs on ossec.log
+file_monitor_timeout = 30
+
+# Expected message to be used on the "callback_macos_uls_log" callback
+expected_message = logcollector.format_macos_message_pattern(macos_message['command'], macos_message['message'])
+
+wazuh_log_monitor = None
+
+
+# Fixtures
+@pytest.fixture(scope='module')
+def startup_cleanup():
+    """Truncate ossec.log and remove logcollector's file_status.json file."""
+    truncate_file(WAZUH_LOCAL_INTERNAL_OPTIONS)
+
+
+@pytest.fixture(scope='module', params=configurations, ids=configuration_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture(scope='module')
+def restart_required_logcollector_daemons():
+    """Wazuh logcollector daemons handler."""
+
+    required_logcollector_daemons = ['wazuh-logcollector']
+
+    for daemon in required_logcollector_daemons:
+        control_service('stop', daemon=daemon)
+
+    truncate_file(LOG_FILE_PATH)
+    os.remove(LOGCOLLECTOR_FILE_STATUS_PATH) if os.path.exists(LOGCOLLECTOR_FILE_STATUS_PATH) else None
+
+    for daemon in required_logcollector_daemons:
+        control_service('start', daemon=daemon)
+
+    yield
+
+    for daemon in required_logcollector_daemons:
+        control_service('stop', daemon=daemon)
```

Comment:
This is needed because of how logcollector when updating the file_status.json, we need to be sure that this file is cleaned before logcollector starts. If we use the 'daemons_handler' fixture, we cannot guarantee that, if we clean the file, it won't change before stoping/restarting logcollector, leading the test to fail.